### PR TITLE
Deprecation fixes and other updates

### LINKED ISSUE #52: Specialize TokenStreamer on the IO type
I did some experiments with the parse Base benchmark and by specializing the `TokenStreamer` on the IO type by adding another type parameter I seem to go from 60% slower than Base parser -> 40 % slower. I have only tested this on my laptop in a VM so it would be cool if someone could reproduce.

The changes to the code needed are trivial (add another parameter to the type and fixup the constructors below).

### PR DESCRIPTION
**Pull request by KristofferC**
Before:
`Parser is ~64.83% slower than base`

After:
`Parser is ~13.38% slower than base`

Parses correctly the same amount of files in Base as before and passes the same amount of tests.

Fixes #52

### CODE REVIEW

File: src/token.jl [hunk 1]
```diff
@@ -118,8 +118,6 @@ function sortedcomplement(of::SourceRange, set)
 end
 
 ⨳(sym::Symbol) = Expr(sym)
-⨳(sym::Symbol, args::LineNumberNode...) = Expr(sym::Symbol, args...)
```
Comment (jakebolewski) [inline src/token.jl]: @KristofferC these my not be exercised in the tests but they may be used in the ASTInterpreter.jl source code, have you checked this?

**Comment by jakebolewski** [comment]
Nice!  Glad to see that even with tracking the source information performance is still competitive.

File: src/token.jl (hunk 1 above)
Comment (KristofferC) [inline src/token.jl]: No, I will reinstate them.

Comment (KristofferC) [inline src/token.jl]: It feels impossible to guard against breaking other packages that rely on unexported and untested methods but I can just add these back, no problems.

**Comment by KristofferC** [comment]
Do you think the type instability of Token has a major performance impact? One way of defining a Token could be with just a string field and a "Kind" field that says what type of Token it is. Kind would typically just be an Enum listing all the different Token types that exist.

File: src/token.jl (hunk 1 above)
Comment (KristofferC) [inline src/token.jl]: Ok, they are put back now.

**Comment by jakebolewski** [comment]
Hmm, I wouldn't want to speculate on the performance diff before trying it first but seems worth trying and shouldn't be too hard.

This LGTM.

**Pull request merged**